pkg/validate/dynamic/loadbalancerprofile.go

```diff
@@ -0,0 +1,102 @@
+package dynamic
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+func (dv *dynamic) ValidateLoadBalancerProfile(ctx context.Context, oc *api.OpenShiftCluster) error {
+	dv.log.Print("ValidateLoadBalancerProfile")
+
+	if oc.Properties.NetworkProfile.OutboundType == api.OutboundTypeUserDefinedRouting {
+		return nil
+	}
+
+	if oc.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs != nil {
+		err := dv.validatePublicIPQuota(ctx, oc)
+		if err != nil {
+			return err
+		}
+	}
+
+	err := dv.validateOBRuleV4FrontendPorts(ctx, oc)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// public IP quota is also checked on the frontend, but only during cluster creation
+func (dv *dynamic) validatePublicIPQuota(ctx context.Context, oc *api.OpenShiftCluster) error {
+	dv.log.Print("validatePublicIPQuota")
+
+	requestedIPs := oc.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs.Count
+
+	if oc.Properties.ProvisioningState == api.ProvisioningStateCreating && oc.Properties.IngressProfiles[0].Visibility == api.VisibilityPublic {
+		requestedIPs += 1
+	} else if oc.Properties.ProvisioningState == api.ProvisioningStateUpdating {
+		currentIPs := len(oc.Properties.NetworkProfile.LoadBalancerProfile.EffectiveOutboundIPs)
+		if requestedIPs > currentIPs {
+			requestedIPs = requestedIPs - currentIPs
+		}
+	}
+
+	netUsages, err := dv.spNetworkUsage.List(ctx, oc.Location)
+	if err != nil {
+		return err
+	}
+
+	for _, netUsage := range netUsages {
+		if *netUsage.Name.Value == "PublicIPAddresses" {
+			if int64(requestedIPs) > (*netUsage.Limit - *netUsage.CurrentValue) {
```

Comment:
Validate that pointer dereferencing does not fail from nil pointers

Reply:
From what I can tell digging through docs/code we shouldn't hit any nil references here, the netUsage is present it should have fields populated.

Also we have been running this same operation on the frontend for some time now with out issue. 
 https://github.com/Azure/ARO-RP/blob/f26f8f41d034618b2042002a90e255f21be1fcc5/pkg/frontend/quota_validation.go#L90-L100